plexpy/notifiers.py

```diff
@@ -2187,6 +2194,165 @@ def _return_config_options(self):
         return config_option
 
 
+class MAILGUN(Notifier):
+    """
+    Mailgun email notifications
+    """
+    NAME = 'Email'
+    _DEFAULT_CONFIG = {'from_name': 'Tautulli',
+                       'from': '',
+                       'domain': '',
+                       'mailing_list_address': '',
+                       'bcc_recipients': 1,
+                       'api_key': '',
+                       'tags': '',
+                       'tracking': 0,
+                       'tracking_clicks': 0,
+                       'tracking_opens': 0,
+                       }
+
+    def agent_notify(self, subject='', body='', action='', **kwargs):
+        if not self.config['api_key']:
+            logger.error("Tautulli Notifiers :: %s notification failed: %s",
+                         self.NAME, "Missing API key")
+            return False
+        
+        if not self.config['from']:
+            logger.error("Tautulli Notifiers :: %s notification failed: %s",
+                         self.NAME, "Missing sender email address")
+            return False
+        
+        if not self.config['domain']:
+            logger.error("Tautulli Notifiers :: %s notification failed: %s",
+                         self.NAME, "Missing domain")
+            return False
+        
+        if not self.config['mailing_list_address']:
+            logger.error("Tautulli Notifiers :: %s notification failed: %s",
+                         self.NAME, "Missing recipient email address")
+            return False
+        
+        # Get emails addresses from Mailgun API
```

Comment:
Factor getting address into a separate method. `agent_notify` should only handle sending the email.